src/main/java/com/redhat/devtools/lsp4ij/server/definition/launching/UserDefinedEditorBehaviorFeature.java

```diff
@@ -0,0 +1,58 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at https://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package com.redhat.devtools.lsp4ij.server.definition.launching;
+
+import com.intellij.psi.PsiFile;
+import com.redhat.devtools.lsp4ij.client.features.EditorBehaviorFeature;
+import com.redhat.devtools.lsp4ij.client.features.LSPClientFeatures;
+import com.redhat.devtools.lsp4ij.server.definition.ClientConfigurableLanguageServerDefinition;
+import com.redhat.devtools.lsp4ij.server.definition.launching.ClientConfigurationSettings.ClientConfigurationEditorSettings;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * User-defined editor behavior feature.
+ */
+public class UserDefinedEditorBehaviorFeature extends EditorBehaviorFeature {
+
+    public UserDefinedEditorBehaviorFeature(@NotNull LSPClientFeatures clientFeatures) {
+        super(clientFeatures);
+    }
+
+    @Nullable
+    private ClientConfigurationEditorSettings getEditorSettings() {
+        ClientConfigurableLanguageServerDefinition serverDefinition = (ClientConfigurableLanguageServerDefinition) getClientFeatures().getServerDefinition();
+        ClientConfigurationSettings clientConfiguration = serverDefinition.getLanguageServerClientConfiguration();
+        return clientConfiguration != null ? clientConfiguration.editor : null;
+    }
+
+    @Override
+    public boolean isEnableStringLiteralImprovements(@NotNull PsiFile file) {
+        ClientConfigurationEditorSettings editorSettings = getEditorSettings();
+        // Note that this defaults to enabled for user-defined language server definitions
+        return (editorSettings == null) || editorSettings.enableStringLiteralImprovements;
```

Comment:
These are all _enabled by default_ here which is why these don't delegate to `super.*()`.